web/tests/stream_topic_history.test.js

```diff
@@ -469,3 +469,52 @@ test("ask_server_for_latest_topic_data", () => {
     assert.deepEqual(history, ["Topic1"]);
     assert.deepEqual(max_message_id, 102);
 });
+
+test("test_reify_local_echo_message", () => {
+    const stream_id = 55;
+    const topic_name = "topic";
+
+    // Trying to reify when history does not exist.
+    assert.equal(
+        stream_topic_history.reify_message_id_if_available({
+            old_id: 47.01,
+            new_id: 48,
+            stream_id: 12,
+            topic_name,
+        }),
+        false,
+    );
+
+    stream_topic_history.add_message({
+        stream_id,
+        message_id: 45.01,
+        topic_name,
+    });
+
+    assert.equal(
+        stream_topic_history.reify_message_id_if_available({
+            old_id: 45.01,
+            new_id: 46,
+            stream_id,
+            topic_name,
+        }),
+        true,
+    );
```

Comment:
Probably good to also have some asserts of what is returned with a `stream_topic_history.get_max_message_id`  type query; I guess we don't have a function to get it for a single topic, but you could consider adding one just to simplify the tests.